core/src/main/java/org/infinispan/transaction/tm/DummyTransaction.java

```diff
@@ -45,7 +47,7 @@
    private final Xid xid;
    private volatile int status = Status.STATUS_UNKNOWN;
    private final List<Synchronization> syncs;
-   private final List<XAResource> resources;
+   private final Map<XAResource, Integer> resources;
```

Comment:
Hm, I am using java.util.Map#replace(K, V) so those resources cannot be replaced, the most that can happen that the badly-behaved TM would still commit twice..